ext/standard/array.c

```diff
@@ -3372,6 +3375,15 @@ static void php_splice(HashTable *in_hash, zend_long offset, zend_long length, H
 	HT_SET_ITERATORS_COUNT(&out_hash, HT_ITERATORS_COUNT(in_hash));
 	HT_SET_ITERATORS_COUNT(in_hash, 0);
 	in_hash->pDestructor = NULL;
+	
+	if (UNEXPECTED(GC_DELREF(in_hash) == 0)) {
+		/* Array was completely deallocated during the operation */
+		zend_array_destroy(in_hash);
```

Comment:
The default branch uses `zend_hash_destroy(in_hash);`. `zend_array_destroy()` seems slightly more optimized (or at least optimized in different ways), but we should stay consistent. Adjusting the default branch would also be possible, but should be measured and done on master.